### DISCUSSION TITLE
Option to keep some destination files on generation

### DISCUSSION THREAD

**Discussion by Liooo** [go-jet/jet#421] [Ideas]
When customizing generated models following [the doc](https://github.com/go-jet/jet/wiki/FAQ#how-to-change-model-field-type), it'd be nice if the user-added custom models resided in the same packages as the jet model, so that we don't have to discriminate the import path depending on custom or not. But currently the generator deletes the whole destination directory, so any files added under `model/*` are deleted.

What do you think of adding an option to keep some of the files under destination directory, for example:

- an option that receives a list of regex or [Match](https://pkg.go.dev/path/filepath#Match) for `filepath.Glob` of the files to keep (or delete)
- a boolean option to completely turn off the deletion (then let users delete whatever they want before the generation)

**Comment by Liooo** [comment]
@go-jet 

What do you think of the idea?

In my case, I want to use go layer defined custom enum type (that's stored in RDB's string type column) without caring if it's jet generated or user defined. For example:

```
./gen
└── mydb
    └── public
        ├── enum
        │   └── user_subscription_type.go
        ├── model
        |   ├── my_custom_users.go  <=== manually add
        │   └── users.go
        └── table
            └── users.go
```

```go
// my_custom_users.go
package model

type UserType string
const (
	UserTypeAdmin UserType = "admin"
	UserTypeManager UserType = "manger"
)
```

```go
// ===== then we can write:
var users []model.User
SELECT(Users.AllColumns).
FROM(Users).
WHERE(AND(
	Users.Type.EQ(model.UserSubscriptionPlan_Pro), // <= this one is generated by jet from postgres'enum type
	Users.Type.EQ(model.UserTypeAdmin),            // <= manually added enums can be referenced like other enums with model.*
)).
Query(db, &users)

// ===== as opposed to:
var users []model.User
SELECT(Users.AllColumns).
FROM(Users).
WHERE(AND(
	Users.Type.EQ(model.UserSubscriptionPlan_Pro),
	Users.Type.EQ(custom_package_somewhere_else.UserTypeAdmin), // <= we need to be aware of where the enum comes from
)).
Query(db, &users)
```

The generation command will be like:


```
jet -dsn=... 
  -schema=... 
  -path=./.gen 
  -preserveFiles="gen/mydb/pubic/model/my_custom_*" 
  -preserveFiles="gen/mydb/pubic/other_patterns"
```

I'm gonna create a PR if it's looking good.

**Comment by trevlt** [comment]
> Users.Type.EQ(model.UserSubscriptionPlan_Pro),

Shouldn't this be `Users.Type.EQ(enum.UserSubscriptionPlan.Pro)`?

> Users.Type.EQ(custom_package_somewhere_else.UserTypeAdmin), // <= we need to be aware of where the enum comes from

My preference is to have separate packages. Because this way, I have a visual distinction between auto-generated and custom packages.
